applications/MeshingApplication/python_scripts/modelers/convert_linear_tets_to_quadratic_modeler.py

```diff
@@ -0,0 +1,107 @@
+
+        This method checks the model part for compatibility, sets necessary variables, and calls the utility function for conversion. It also handles the replacement of elements and conditions based on provided settings.
+        """
+        super().SetupModelPart()
+
+        if KratosMultiphysics.IsDistributedRun():
+            raise Exception("ConvertLinearTetsToQuadraticModeler is not MPI-ready")
+
+        model_part = self.model[self.settings["model_part_name"].GetString()]
+
+        for elem in model_part.Elements:
+            if elem.GetGeometry().GetGeometryType() != KratosMultiphysics.GeometryData.KratosGeometryType.Kratos_Tetrahedra3D4:
+                raise Exception("Only linear tets are accepted as input (4 nodes)")
+
+        KratosMultiphysics.VariableUtils().SetNonHistoricalVariable(
+            KratosMultiphysics.SPLIT_ELEMENT,
+            True,
+            model_part.Elements)
+        KratosMultiphysics.VariableUtils().SetNonHistoricalVariable(
+            KratosMultiphysics.SPLIT_ELEMENT,
+            True,
+            model_part.Conditions)
+
+        Meshing.LinearToQuadraticTetrahedraMeshConverter(model_part.GetRootModelPart()).LocalConvertLinearToQuadraticTetrahedraMesh(False, False)
+
+        element_name = self.settings["element_name"].GetString()
+        if element_name:
+            replace_settings = KratosMultiphysics.Parameters("{}")
+            replace_settings.AddString("element_name", element_name)
+            replace_settings.AddString("condition_name", "")
+            KratosMultiphysics.ReplaceElementsAndConditionsProcess(model_part, replace_settings).Execute()
+
+    def __GetDefaultSettings(self):
+        """
+        Provides the default settings for the modeler.
+
+        Returns:
+            KratosMultiphysics.Parameters: The default settings as a Parameters object.
+
+        These settings include the name of the model part to be modified and the name of the element to use for replacement, if specified.
+        """
+        return KratosMultiphysics.Parameters('''{
+            "model_part_name" : "main_model_part",
+            "element_name" : ""
+        }''')
+
+    def __str__(self):
+        """
+        String representation of the ConvertLinearTetsToQuadraticModeler class.
+
+        Returns:
+            str: A string describing the class.
+        """
+        return "ConvertLinearTetsToQuadraticModeler"
+
+def Factory(model, settings):
+    """
+    Factory method to create an instance of ConvertLinearTetsToQuadraticModeler.
+
+    Args:
+        model (KratosMultiphysics.Model): The Kratos model to be modified.
+        settings (KratosMultiphysics.Parameters): Configuration settings for the modeler.
+
+    Returns:
+        ConvertLinearTetsToQuadraticModeler: An instance of the modeler class.
+    """
+    return ConvertLinearTetsToQuadraticModeler(model, settings)
```

Comment:
As this is a new feature, I'd avoid adding the factory method to promote the use of the registry (this of course requires adding this class to the Python layer of the registry).

Reply:
I think everything is correct. Thanks!